zookeeper-server/src/main/java/org/apache/zookeeper/server/persistence/FileTxnSnapLog.java

```diff
@@ -595,7 +600,9 @@ public void rollLog() throws IOException {
      */
     public void close() throws IOException {
         txnLog.close();
+        txnLog = null;
```

Comment:
if you set this variable to null then calling twice this method will result in an ugly NullPointerException
what about
```
try {
if (txnLog != null) {
      txnLog.close();
}
} finally {
txnLog = null;
}

```

Reply:
Done